### PR TITLE
[opentelemetry-cpp-contrib-version] Add port opentelemetry-cpp-contrib-version

### PR DESCRIPTION
**Pull request by Pravila00**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] <del> The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines. </del> 
- [ ] <del> Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html) </del> 
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [ ] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [ ] <del> The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context. </del>
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by WangWeiLin-MV — DISMISSED** [review]
The port feature `user-events` installation tests pass with the following triplets:
* x64-linux

**Comment by BillyONeal** [comment]
In https://github.com/microsoft/vcpkg/pull/37368/ we said we were not comfortable meaningfully changing the upstream sources that are installed with the version number not changing like this. (I realize now that in https://github.com/microsoft/vcpkg/pull/37966 we messed this up; that somehow got missed in review)

I think either upstream needs to come up with a versioning strategy for their software that makes sense for customers to rely on, or we need to give up and switch to `"version-date"`.

**Comment by Pravila00** [comment]
@BillyONeal I understand your point here. 

I think the best way to proceed would be creating a new port opentelemetry-cpp-contrib. I can target that. Nonetheless, could we merge this PR now and work on that in a different PR?

We really need to have these changes for business purposes, so being blocked here is really painful at the moment. What do you think?

**Comment by BillyONeal** [comment]
@AugP
@JavierMatosD
@ras0219-msft
and I discussed today.

@ras0219-msft has a suggestion in addition to my two above. We could add another port `opentelementy-cpp-contrib-version` which just publishes the ref and sha as a CMake helper port (like `vcpkg-cmake`). Then it can have its own versioning.

That is, extract this block into a separate function which gets published as a helper:

https://github.com/microsoft/vcpkg/blob/e3ad383ceb886677eb181e16adeeb4d4a2c2d4c4/ports/opentelemetry-cpp/portfile.cmake#L51-L59

@Pravila00 can you make a change like that?

**Comment by Pravila00** [comment]
Change is almost ready, I am waiting for #39821 to be fixed.

**Comment by Pravila00** [comment]
This is ready for review again, please test that "**user-events**" and "**geneva**" features continue to work fine.

File: ports/opentelemetry-cpp-contrib-version/clone_opentelemetry_cpp_contrib.cmake
Comment (WangWeiLin-MV) [inline ports/opentelemetry-cpp-contrib-version/clone_opentelemetry_cpp_contrib.cmake:1]: Please rename filename to `vcpkg-port-config.cmake`.

And unified the EOL to `LF`.

File: ports/opentelemetry-cpp-contrib-version/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+file(INSTALL
+    "${CMAKE_CURRENT_LIST_DIR}/clone_opentelemetry_cpp_contrib.cmake"
```
Comment (WangWeiLin-MV) [inline ports/opentelemetry-cpp-contrib-version/portfile.cmake:2]:
```suggestion
    "${CMAKE_CURRENT_LIST_DIR}/vcpkg-port-config.cmake"
```

File: ports/opentelemetry-cpp/portfile.cmake [hunk 2]
```diff
@@ -50,13 +50,8 @@ set(OPENTELEMETRY_CPP_EXTERNAL_COMPONENTS "OFF")
...
     # Geneva and user events exporters from opentelemetry-cpp-contrib are tightly coupled with opentelemetry-cpp repo, 
     # so they should be ported as a feature under opentelemetry-cpp.
-    vcpkg_from_github(
-        OUT_SOURCE_PATH CONTRIB_SOURCE_PATH
-        REPO open-telemetry/opentelemetry-cpp-contrib
-        REF 4f3059390ad09d12d93255a10f7be8ff948d26fc
-        HEAD_REF main
-        SHA512 392e3a414ea0ee016768dd75a286d2a7a3a7a011cc0fce4ee2f796ad7e0cd70d534eb38cb1d22f91300f4670dfa212a4239ba8e008c2444d47e13c5fe3fb75c0
-    )
+    include(${CURRENT_HOST_INSTALLED_DIR}/share/opentelemetry-cpp-contrib-version/clone_opentelemetry_cpp_contrib.cmake)
```
Comment (WangWeiLin-MV) [inline ports/opentelemetry-cpp/portfile.cmake:53]:
```suggestion
```

File: ports/opentelemetry-cpp-contrib-version/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (WangWeiLin-MV) [inline ports/opentelemetry-cpp-contrib-version/portfile.cmake:5]: Please append the EOL `LF` for the last line.

File: ports/opentelemetry-cpp-contrib-version/portfile.cmake (hunk 1 above)
Comment (Pravila00) [reply ports/opentelemetry-cpp-contrib-version/portfile.cmake]: Updated, thanks!

File: ports/opentelemetry-cpp-contrib-version/clone_opentelemetry_cpp_contrib.cmake
Comment (Pravila00) [reply ports/opentelemetry-cpp-contrib-version/clone_opentelemetry_cpp_contrib.cmake:1]: Unified to LF and file renamed

File: ports/opentelemetry-cpp/portfile.cmake (hunk 2 above)
Comment (Pravila00) [reply ports/opentelemetry-cpp/portfile.cmake]: I've removed the "include"

File: ports/opentelemetry-cpp-contrib-version/portfile.cmake (hunk 3 above)
Comment (Pravila00) [reply ports/opentelemetry-cpp-contrib-version/portfile.cmake]: It should be there now, can you double check from your side? Not sure if GitHub code diff is showing that on the UI.

**Review by WangWeiLin-MV — APPROVED** [review]
The port feature `user-events` and `genova` installation tests pass with the following triplets:
* x64-linux

**Pull request merged**